libraries/AP_DDS/AP_DDS_Gather_Data.cpp

```diff
@@ -0,0 +1,20 @@
+#if AP_DDS_ENABLED
+
+#include "AP_DDS_Gather_Data.h"
+#include <AP_HAL/AP_HAL.h>
+#include <AP_RTC/AP_RTC.h>
+
+void update_topic(builtin_interfaces_msg_Time* msg)
```

Comment:
why is update_topic not in the class?
maybe it should be a member of AP_DDS_Client::Topic_table, so we loop calling update_topic method on each of the topic table entries?

Reply:
It can be part of the class for sure. 

As far as in the topic table, I'm not sure how much I want to build into that yet because the we might break the more than one updater per topic assumption soon with, for example, multi battery support.

Reply:
I'm realizing it's not in the class is because we were going to code generate it, but this is the one part that definately can't be generated because it's the bridge to the rest of the API. 

TLDR: It's in the class now, in commit 61099c35ac0dc135baa42d4179bbb0f6fedc58ac